### PR TITLE
CB2-9616: Open Api Spec

### PR DESCRIPTION
**Pull request by Daniel-Searle**
## Add missing API spec to document retrieval
This change is to add in the mission open api spec yml to the project.
[CB2-9616](https://dvsa.atlassian.net/browse/CB2-9616)

## Checklist
- [ ] Code has been tested manually
- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: docs/spec.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
+openapi: 3.0.0
```
Comment (cb-cs) [inline docs/spec.yml:1]: Is it worth adding the following to the .gitIgnore file for those of us that use IDEA IDE's?

# Idea specific hidden files
.idea/**

When I am running this locally, it is prompting me to commit.

File: docs/spec.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  /document-retrieval:
+    get:
+      summary: Retrieve a Document
+      parameters:
+        - name: vinNumber
+          in: query
+          description: VIN number
+          required: false
+          schema:
+            type: string
```
Comment (cb-cs) [inline docs/spec.yml:29]: Should there also be some sort of reference here to the required regex/pattern as per docs?

https://swagger.io/docs/specification/data-models/data-types/#pattern

File: docs/spec.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        - name: testNumber
+          in: query
+          description: Test number
+          required: false
+          schema:
+            type: string
```
Comment (cb-cs) [inline docs/spec.yml:35]: Should there also be some sort of reference here to the required regex/pattern as per docs?

https://swagger.io/docs/specification/data-models/data-types/#pattern

File: docs/spec.yml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        - name: plateSerialNumber
+          in: query
+          description: Plate serial number
+          required: false
+          schema:
+            type: string
```
Comment (cb-cs) [inline docs/spec.yml:41]: Should there also be some sort of reference here to the required regex/pattern as per docs?

https://swagger.io/docs/specification/data-models/data-types/#pattern

File: docs/spec.yml [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        - name: systemNumber
+          in: query
+          description: System number
+          required: false
+          schema:
+            type: string
```
Comment (cb-cs) [inline docs/spec.yml:47]: Should there also be some sort of reference here to the required regex/pattern as per docs?

https://swagger.io/docs/specification/data-models/data-types/#pattern

File: docs/spec.yml (hunk 3 above)
Comment (LGin-BJSS) [reply docs/spec.yml:35]: I would have the pattern here for simple params, but lower down if defined as a reference type.

File: docs/spec.yml (hunk 4 above)
Comment (LGin-BJSS) [reply docs/spec.yml:41]: I would have the pattern here for simple params, but lower down if defined as a reference type.

File: docs/spec.yml (hunk 5 above)
Comment (LGin-BJSS) [reply docs/spec.yml:47]: I would have the pattern here for simple params, but lower down if defined as a reference type.

**Review by LGin-BJSS — APPROVED** [review]

**Review by cb-cs — APPROVED** [review]

**Pull request merged**